apps/examples/testcase/ta_tc/ble_manager/itc/itc_blemanager_main.c

```diff
@@ -0,0 +1,865 @@
+/****************************************************************************
+ *
+ * Copyright 2022 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/time.h>
+#include <ble_manager/ble_manager.h>
+#include <semaphore.h>
+#include <stdbool.h>
+#include <errno.h>
+#include "tc_common.h"
+
+#define LOOP_SIZE 2	//used to compute average time for different operations
+
+#define ITC_FUNC_SIGNAL              \
+	do {                                 \
+		sem_post(&g_func_sem);        \
+	} while (0)
+
+#define ITC_FUNC_WAIT                \
+	do {                                 \
+		sem_wait(&g_func_sem);        \
+	} while (0)
+
+static sem_t g_func_sem;
+
+static ble_addr g_target = { 0, };
+
+static uint8_t ble_filter[] = { 0x02, 0x01, 0x05, 0x03, 0x19, 0x80, 0x01, 0x05, 0x03, 0x12, 0x18, 0x0f, 0x18 };
+
+static void addr_string_to_mac(const char *address)
+{
+        int i;
+        char *ptr = NULL;
+
+        for (i = 0; i < BLE_BD_ADDR_MAX_LEN; i++) {
+                g_target.mac[i] = strtol(address, &ptr, 16);
+                if (ptr != NULL) {
+                        if (ptr[0] != ':')
+                                return;
+                        address = ptr + 1;
+                }
+        }
+}
+
+static void ble_scan_state_changed_cb(ble_scan_state_e scan_state)
```

Comment:
return is unnecessary for void type. please check all other function

Reply:
Done